cirq-core/cirq/ops/state_preparation_channel_test.py

```diff
@@ -0,0 +1,128 @@
+    gate = cirq.StatePreparationChannel(state)(qubits[0], qubits[1])
+    state = state / np.linalg.norm(state)
+    np.testing.assert_almost_equal(
+        cirq.kraus(gate),
+        (
+            np.array([state, np.zeros(4), np.zeros(4), np.zeros(4)]).T,
+            np.array([np.zeros(4), state, np.zeros(4), np.zeros(4)]).T,
+            np.array([np.zeros(4), np.zeros(4), state, np.zeros(4)]).T,
+            np.array([np.zeros(4), np.zeros(4), np.zeros(4), state]).T,
+        ),
+    )
+
+
+def test_state_prep_channel_kraus_small():
+    gate = cirq.StatePreparationChannel(np.array([0.0, 1.0]))(cirq.LineQubit(0))
+    np.testing.assert_almost_equal(
+        cirq.kraus(gate), (np.array([[0.0, 0.0], [1.0, 0.0]]), np.array([[0.0, 0.0], [0.0, 1.0]]))
+    )
+    assert cirq.has_kraus(gate)
+    assert not cirq.has_mixture(gate)
+
+    gate = cirq.StatePreparationChannel(np.array([1.0, 0.0]))(cirq.LineQubit(0))
+    np.testing.assert_almost_equal(
+        cirq.kraus(gate), (np.array([[1.0, 0.0], [0.0, 0.0]]), np.array([[0.0, 1.0], [0.0, 0.0]]))
+    )
+    assert cirq.has_kraus(gate)
+    assert not cirq.has_mixture(gate)
+
+
+def test_state_prep_gate_printing():
+    circuit = cirq.Circuit()
+    qubits = cirq.LineQubit.range(2)
+    gate = cirq.StatePreparationChannel(np.array([1, 0, 0, 1]) / np.sqrt(2))
+    circuit.append(cirq.H(qubits[0]))
+    circuit.append(cirq.CNOT(qubits[0], qubits[1]))
+    circuit.append(gate(qubits[0], qubits[1]))
+    cirq.testing.assert_has_diagram(
+        circuit,
+        """
+0: ───H───@───StatePreparation[1]───
+          │   │
+1: ───────X───StatePreparation[2]───
+""",
+    )
+
+
+@pytest.mark.parametrize('name', ['Prep', 'S'])
+def test_state_prep_gate_printing_with_name(name):
+    circuit = cirq.Circuit()
+    qubits = cirq.LineQubit.range(2)
+    gate = cirq.StatePreparationChannel(np.array([1, 0, 0, 1]) / np.sqrt(2), name=name)
+    circuit.append(cirq.H(qubits[0]))
+    circuit.append(cirq.CNOT(qubits[0], qubits[1]))
+    circuit.append(gate(qubits[0], qubits[1]))
+    cirq.testing.assert_has_diagram(
+        circuit,
+        f"""
+0: ───H───@───{name}[1]───
+          │   │
+1: ───────X───{name}[2]───
+""",
+    )
+
+
+def test_gate_params():
+    state = np.array([1, 0, 0, 0], dtype=np.complex64)
+    gate = cirq.StatePreparationChannel(state)
+    assert gate.num_qubits() == 2
+    assert not gate._has_unitary_()
+    assert gate._has_kraus_()
+    assert (
+        repr(gate)
+        == 'cirq.StatePreparationChannel(np.array([(1+0j), 0j, 0j, 0j], dtype=np.complex128))'
+    )
+
+
+def test_gate_error_handling():
+    with pytest.raises(ValueError, match='`target_state` must be a 1d numpy array.'):
+        cirq.StatePreparationChannel(np.eye(2))
+    with pytest.raises(ValueError, match=f'Matrix width \\(5\\) is not a power of 2'):
+        cirq.StatePreparationChannel(np.ones(shape=5))
+
+
+def test_equality_of_gates():
+    state = np.array([1, 0, 0, 0], dtype=np.complex64)
```

Comment:
Use `EqualsTester` instead to check for equality. It will also help check that two `StatePreparationChannel`s with different `state`s are not equal.